contrib/NeonVM/neonvm.c

```diff
@@ -217,510 +229,523 @@ static void loadBytecode(const unsigned char *bytecode, unsigned int len, struct
 int main(int argc, char* argv[])
 {
     if (argc <= 1) {
-        printf("\nusage: %s program.neon\n", argv[0]);
+        printf("\nusage: %s program.neonx\n", argv[0]);
         return 1;
     }
     struct tagTBytecode *pModule;
     FILE *fp = fopen(argv[1], "rb");
     fseek(fp, 0, SEEK_END);
-    long nSize = ftell(fp);
+    size_t nSize = ftell(fp);
     fseek(fp, 0, SEEK_SET);
 
-    pModule = (struct tagTBytecode *)malloc((sizeof(struct tagTBytecode)));
-    unsigned char *bytecode = (unsigned char*)malloc(nSize);
+    pModule = malloc((sizeof(struct tagTBytecode)));
+    if (pModule == NULL) {
+        fatal_error("Could not allocate memory for neon executable.  Error %s", _strerror(NULL));
+    }
+    uint8_t *bytecode = malloc(nSize);
+    if (bytecode == NULL) {
+        fatal_error("Could not allocate memory for neon bytecode.  Error %s.", _strerror(NULL));
+    }
     size_t bytes_read = fread(bytecode, 1, nSize, fp);
 
     loadBytecode(bytecode, bytes_read, pModule);
 
     struct tagTExecutor *exec = new_executer(pModule);
-    exec->exec_run(exec);
-
+    exec_run(exec);
 }
 
 void exec_run(struct tagTExecutor *self)
 {
-    push(&self->callstack, &self->object->codelen);
+    push(self->callstack, &self->object->codelen);
     exec_loop(self);
 
-    for (unsigned int i = 0; i < self->object->strtablelen; i++) {
+    for (uint32_t i = 0; i < self->object->strtablelen; i++) {
         free(self->object->strtable[i]);
     }
 }
 
 struct tagTExecutor *new_executer(struct tagTBytecode *object)
 {
     struct tagTExecutor *r = malloc(sizeof(struct tagTExecutor));
+    if (r == NULL) {
+        fatal_error("Failed to allocate memory for Neon executable.  Error: ", _strerror(NULL));
+    }
     r->object = object;
-    r->callstack = NULL;
-    r->stack = NULL;
+    r->callstack = createStack(300);
+    r->stack = createStack(300);
     r->ip = 0;
-    r->exec_loop = exec_loop;
-    r->exec_run = exec_run;
+    return r;
+}
+
+static uint32_t exec_getOperand(struct tagTExecutor *self)
+{
+    uint32_t r = (self->object->code[self->ip+1] << 24) | (self->object->code[self->ip+2] << 16) | (self->object->code[self->ip+3] << 8) | self->object->code[self->ip+4];
+    self->ip += 5;

```

Comment:
This is probably going to cause a problem because you're only consuming 4 bytes here, not 5. Some opcodes have two operands.